packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -140,18 +157,26 @@ export const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
   const schemaFields = useAutocompleteFields(namespace);
 
   const completer = useMemo(() => {
-    return createQueryAutocompleter({
-      fields: schemaFields,
-      serverVersion,
-    });
-  }, [schemaFields, serverVersion]);
+    return createQueryWithHistoryAutocompleter(
+      savedQueries.map((query) => ({
```

Comment:
Are we accounting for batch operations somehow here? Maybe we should filter those out? (totally not sure it's the way to go, just wondering) 🤔